suborbital/atmo
# Add support for configuring auth provider in Directive

This PR exposes Reactr's AuthProvider as config in the Directive. It allows you to set auth headers for particular domains such that GraphQL and HTTP calls made by Runnables get the appropriate headers to authenticate.

This includes the ability to get auth values from the environment:
```yaml
authentication:
  domains:
    api.github.com:
      headertype: bearer
      value: env(GITHUB_TOKEN)
```

To test this PR:

- build the example project: `subo build ./example-project`
- Get a GitHub personal access token from your account and run `export GITHUB_TOKEN={your token}`
- Run `make atmo bundle=./example-project/runnables.wasm.zip` and then call GET /github

The output should be some information about the Reactr repo (GitHub's GraphQL endpoint requires authentication, and it's being provided by the directive configuration)

Closes https://github.com/suborbital/atmo/issues/34
